### PR TITLE
Hardhat basic test setup

### LINKED ISSUE #81: Setup testing environment
### Goal
(Manual) testing environment

### Context
As we're experiencing problems testing auctions on the test chains (even manually) we want to have a possibility to test our code on the mainnet fork. This potentially will require:

- Setting up hardhat
  - Registering `alchemy` account, since `infura` doesn't provide historical info
  - Configuring it to fork mainnet https://hardhat.org/hardhat-network/guides/mainnet-forking.html
- Pointing frontend to the new RPC endpoint
- Setting up [ethernal](https://doc.tryethernal.com/getting-started/quickstart) or something similar to view local transactions
- Writing a script to create a vault, change medianizer price, bark – to start an auction

Note: as we're waiting for the Chaoslabs to provide us access their testing tool, some of the points from above may be dropped.

### Tasks
- [ ] investigate and document our approach to testing
- [ ] update list of tasks

### PR DESCRIPTION
**Pull request by valiafetisov**
Closes #81 

This PR introduces very basic setup for the manual testing, at the moment it only forks mainnet at a specific block, where few auctions are already running. It also fixes some problems with working with forks such as fetching "current date" from a blockchain in order to display correct estimations.

Note: ~~currently nothing works beyond displaying auctions, but it's still useful to start on the #73~~
Participation also works now: you can execute auth and bid on the profitable auctions using one of the hardhat accounts.

Screenshot 2022-02-08 at 15 17 56

Testing scenario is described in the [hardhat/README.md](https://github.com/sidestream-tech/unified-auctions-ui/blob/hardhat-testing-fixes/hardhat/README.md)

### CODE REVIEW

**Comment by LukSteib** [comment]
Where do I get the `Alchemy URL` env variable from?

> (can be found in: apps table -> app -> view key)

What does this refer to?

**Comment by valiafetisov** [comment]
The one I've created is stored in the `secret export --format=dotenv auction-ui/localhost/hardhat` as per readme 🙂

**Comment by LukSteib** [comment]
* was able to get the setup started ✔️ 
* auctions are fetched ✔️ 

How do I use the hardhat accounts for execution?

**Comment by valiafetisov** [comment]
> How do I use the hardhat accounts for execution?

When the hardhat is started it prints a list of account addresses and their private keys. You can take any of those private keys and add it to the Metamask:
- Click on the metamask account icon
- Select `Import Account`
- Paste the key
- Select the new account, while being connected to the frontend
- You can now execute transactions

Fun part: people sometimes mix up those testing accounts with their own ones and [send there real money](https://etherscan.io/address/0xf39fd6e51aad88f6f4ce6ab8827279cfffb92266). Therefore, _be careful, never mix up testing metamask profile with the accounts that holds actual money_.

**Comment by LukSteib** [comment]
* Authorisation worked ✔️ 
* Not able to execute an auction (tested with `ETH-C` and LP token auction). Getting the following callee related error:
```
Bidding error Error: Can not find callee addresses for the "localhost" network
    at getCalleesByNetworkType (CALLEES.ts?96e2:24:1)
    at getCalleeAddressByCollateralType (CALLEES.ts?96e2:30:1)
    at _callee8$ (auctions.ts?450c:147:1)
    at tryCatch (runtime.js?96cf:62:1)
    at Generator.invoke [as _invoke] (runtime.js?96cf:296:1)
    at Generator.prototype.<computed> [as next] (runtime.js?96cf:114:1)
    at asyncGeneratorStep (asyncToGenerator.js?1da1:3:1)
    at _next (asyncToGenerator.js?1da1:25:1)
    at eval (asyncToGenerator.js?1da1:32:1)
    at new Promise (<anonymous>)
```

**Comment by valiafetisov** [comment]
> Getting the following callee related error

Please pull latest changes since it was implemented in a066bbb, after I pinged for the review.

**Review by LukSteib — DISMISSED** [review]

**Comment by LukSteib** [comment]
* Executed several auctions ✔️

File: core/src/time.ts [hunk 1]
```diff
@@ -0,0 +1,39 @@
+import memoizee from 'memoizee';
```
Comment (BracketJohn) [inline core/src/time.ts:1]: The naming of this module + its main export `getCurrentDate` could be clearer - without enough context it's not clear what _current_ means here

**Review by BracketJohn — COMMENTED** [review]
Happy that we can test locally now 🎊 

I only looked over the code briefly, as @LukSteib already got it to run. Maybe another team member should try to get it to run before merging it?

**Review by BracketJohn — APPROVED** [review]

**Pull request merged**